src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayoutSpec.scala

```diff
@@ -438,6 +439,55 @@ class hmrcLayoutSpec extends AnyWordSpecLike with Matchers with JsoupHelpers wit
       backLink.html()       shouldBe "Back Welsh"
     }
 
+    "include the back link if passed a BackLink" in {
+      val hmrcLayout     = app.injector.instanceOf[HmrcLayout]
+      val messages       = getMessages()
+      val customBackLink = BackLink("/some/url", "custom-class", Map("some" -> "attribute"), Text("Custom Back"))
+      val layout         = hmrcLayout(backLink = Some(customBackLink))(
+        Html("")
+      )(fakeRequest, messages)
+      val document       = Jsoup.parse(contentAsString(layout))
+
+      val actualBackLink = document.select(".govuk-back-link")
+      actualBackLink                            should have size 1
+      actualBackLink.attr("href")             shouldBe "/some/url"
+      actualBackLink.hasClass("custom-class") shouldBe true
+      actualBackLink.attr("some")             shouldBe "attribute"
+      actualBackLink.html()                   shouldBe "Custom Back"
+    }
+
+    "add attributes to activate the JavaScript browser back in hmrc-frontend" in {
+      val hmrcLayout                  = app.injector.instanceOf[HmrcLayout]
+      implicit val messages: Messages = getMessages()
+      val layout                      = hmrcLayout(backLink = Some(BackLink.mimicsBrowserBackButtonViaJavaScript))(
+        Html("")
+      )(fakeRequest, messages)
+      val document                    = Jsoup.parse(contentAsString(layout))
+
+      val actualBackLink = document.select(".govuk-back-link")
+      actualBackLink                       should have size 1
+      actualBackLink.attr("href")        shouldBe "#"
+      actualBackLink.attr("data-module") shouldBe "hmrc-back-link"
+      actualBackLink.html()              shouldBe "Back English"
+    }
+
+    "give backLink precedence over backLinkUrl" in {
```

Comment:
I'm assuming this is what we want, rather than throwing an exception or showing both?

Reply:
sorry for the delay, meetings interrupted me. Maybe we should @require that we only get one or the other, I could imagine that people might pass both by mistake and have a wrong assumption about how it will behave that they don't notice. Since backLink is new it doesn't feel dangerous to do, and we've used it in other components to guard against stuff https://github.com/hmrc/play-frontend-hmrc/blob/8969a6a0a56d1b185b9d67bfa603b07f5bd28170/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCharacterCount.scala.html#L23